### PR TITLE
chore: re-org pipelines

### PR DESCRIPTION
**Pull request by PsiACE**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Just a few file moves and text replacements.

**before**

```
.
├── *.rs
├── new/
│   ├── executor/
│   ├── processors/
│   │   ├── *.rs
│   │   ├── sinks/
│   │   ├── sources/
│   │   └── transforms/
│   └──  *.rs
└── transforms/
    ├── group_by/
    └── *.rs
```

**after**

```
.
├── executor/
├── processors/
│   ├── *.rs
│   ├── sinks/
│   ├── sources/
│   └── transforms/
└── *.rs
```

Fixes #issue

### CODE REVIEW

**Comment by BohuTANG** [comment]
This may affect patch https://github.com/datafuselabs/databend/pull/6440, shall we do this after it merged? @leiysky

**Comment by Xuanwo** [comment]
`test_metactl (pull_request)` could be resolved by merge with main.

**Comment by Xuanwo** [comment]
@mergify update

**Comment by leiysky** [comment]
> This may affect patch #6440, shall we do this after it merged? @leiysky

It won't affect #6440 too much, we can merge this now.

**Review by leiysky — APPROVED** [review]
LGTM, @zhang2014 do we have plan to remove the legacy transforms recently?

File: query/src/api/rpc/exchange/exchange_sink.rs [hunk 1]
```diff
@@ -21,10 +21,10 @@ use crate::api::rpc::exchange::exchange_params::ExchangeParams;
 use crate::api::rpc::exchange::exchange_params::MergeExchangeParams;
 use crate::api::rpc::exchange::exchange_sink_merge::ExchangeMergeSink;
 use crate::api::rpc::exchange::exchange_sink_shuffle::ExchangePublisherSink;
-use crate::pipelines::new::processors::port::InputPort;
-use crate::pipelines::new::processors::processor::ProcessorPtr;
-use crate::pipelines::new::NewPipeline;
-use crate::pipelines::new::SinkPipeBuilder;
+use crate::pipelines::processors::port::InputPort;
+use crate::pipelines::processors::processor::ProcessorPtr;
+use crate::pipelines::NewPipeline;
```
Comment (Xuanwo) [inline query/src/api/rpc/exchange/exchange_sink.rs:26]: How about renaming to `Pipeline`

File: query/src/catalogs/hive/hive_table.rs [hunk 2]
```diff
@@ -34,13 +34,13 @@ use opendal::ObjectMode;
...
-use crate::pipelines::new::processors::SyncSource;
-use crate::pipelines::new::processors::SyncSourcer;
-use crate::pipelines::new::NewPipe;
-use crate::pipelines::new::NewPipeline;
-use crate::pipelines::new::SourcePipeBuilder;
+use crate::pipelines::processors::port::OutputPort;
+use crate::pipelines::processors::processor::ProcessorPtr;
+use crate::pipelines::processors::SyncSource;
+use crate::pipelines::processors::SyncSourcer;
+use crate::pipelines::NewPipe;
```
Comment (Xuanwo) [inline query/src/catalogs/hive/hive_table.rs:41]: ditto

File: query/src/catalogs/hive/hive_table.rs [hunk 3]
```diff
@@ -34,13 +34,13 @@ use opendal::ObjectMode;
...
+use crate::pipelines::NewPipeline;
```
Comment (Xuanwo) [inline query/src/catalogs/hive/hive_table.rs:42]: ditto

**Review by zhang2014 — APPROVED** [review]

**Comment by zhang2014** [comment]
> GTM, @zhang2014 do we have plan to remove the legacy transforms recently?

needs some refactoring(format sources)

**Pull request merged**